### DISCUSSION TITLE
localization in multi-tenant application

### DISCUSSION THREAD

**Discussion by tylkomat** [payloadcms/payload#5685] [Q&A]
Is it possible in a multi-tenant-application to have localization based on the tenant language?
Each tenant should be able to localize fields based on english and their local language. In case english is their local language there should be only one option.

**Comment by DanRibbens** [comment]
Yes, Payload makes it possible to define all kinds of localization.

This entirely depends on your frontend app to request the locales. When using the rest of graphql APIs you'll be able to pass the `?locale=en&fallbackLocale=es` for example or in graphql, the locale is an argument.

If you wanted to make is so that only certain tenants had certain locales there would be some complexity there based on how Payload handles locales currently. We've discussed in the future having access control on specific locales, this would be needed to prevent tenants from making content in every language, but I'm not sure if that is really what you're asking here.

There are all kinds of different localization examples for whatever frontend you're building with, without knowing what you're setting up, I can't answer this part of your question.

Does this clear things up for you?

**Reply by tylkomat** [reply]
Thank you for the detailed answer and the hint about the frontend. My questions are regarding the backend, meaning payload itself.

> If you wanted to make is so that only certain tenants had certain locales there would be some complexity there based on how Payload > handles locales currently. We've discussed in the future having access control on specific locales, this would be needed to prevent tenants from making content in every language, but I'm not sure if that is really what you're asking here.

Yes that is exactly what I'm asking.
I currently have like 20 languages defined. This makes the language selector in the backend huge. Each Tenant should basically just see two languages. Could this be handled with a custom language selector component or is it not possible to replace that one?

I also would like to have the fallback language set to the default language of each tenant. English should be the secondary language. For example a tenant from germany would have german as fallback language, since all created content will definitely be in german, some content may be also in english as secondary option. Similar for a polish tenant. All content will definitely be in polish, but some may additionally be in english.
It looks like payload lets me only set the fallback language globally.

**Reply by DanRibbens** [reply]
We haven't made the locale selector a swappable component, but there is an easy workaround using admin.compenents.actions.

In your case I would copy the code for the locale selector to add to your own actions. Then you can hide the default one with custom css setting the display: 'none' for that selector.

In your component:
 - change the baseClass so it won't be hidden
 - useAuth to get the user from auth provider
 - limit the locale options based on the user's tenant locales which you may need to get from another fetch or use the depth property on the auth config of your users to get what you need.

This should work fine for your use. Please share your solution here for others too if you can.

**Reply by tylkomat** [reply]
Thank you, I'll try your suggestion.
I saw there are other Discussions about changes to localization. Hopefully they won't break my work :)
